go/worker/compute/executor/committee/node.go

```diff
@@ -424,6 +485,229 @@ func (n *Node) HandleNewBlockLocked(blk *block.Block) {
 	}
 }
 
+// CheckTx checks the given call in the node's runtime.
+func (n *Node) checkTx(ctx context.Context, call []byte) error {
+	n.commonNode.CrossNode.Lock()
+	currentBlock := n.commonNode.CurrentBlock
+	n.commonNode.CrossNode.Unlock()
+
+	if currentBlock == nil {
+		return errNotReady
+	}
+
+	checkRq := &protocol.Body{
+		RuntimeCheckTxBatchRequest: &protocol.RuntimeCheckTxBatchRequest{
+			Inputs: transaction.RawBatch{call},
+			Block:  *currentBlock,
+		},
+	}
+	rt := n.GetHostedRuntime()
+	if rt == nil {
+		n.logger.Error("CheckTx: hosted runtime not initialized")
+		return errNotReady
+	}
+	resp, err := rt.Call(ctx, checkRq)
+	switch {
+	case err != nil:
+		n.logger.Error("CheckTx: runtime call error",
+			"err", err,
+		)
+		return err
+	case resp == nil:
```

Comment:
This can never be nil (if it was `err` would have an error), but the underlying message can be:
```suggestion
	case resp.RuntimeCheckTxBatchResponse == nil:
```